packages/core/src/locator.ts

```diff
@@ -1,8 +1,15 @@
 import { ConstructorOrFactory, ClassConstructor } from './internal';
 
 export type Instantiator<Context> = <T>(f: ConstructorOrFactory<Context, T>) => T;
+export type InterceptorGet<Context> = (i: Instantiator<Context>) => Instantiator<Context>;
 
-export type Interceptor<Context> = (i: Instantiator<Context>) => Instantiator<Context>;
+export interface Interceptor<C> {
+  get?: InterceptorGet<C>;
+  has?: <T>(f: ConstructorOrFactory<C, T>) => boolean;
```

Comment:
ditto, an interceptor cannot have a `has` method because all it does is wrap things